packages/headless/src/app/engine-configuration.ts

```diff
@@ -150,6 +150,12 @@ export interface AnalyticsConfiguration {
    * Specifies the URL of the current page or component.
    */
   documentLocation?: string;
+  /**
+   * TBD
+   * @internal
+   */
+  trackingId?: string;
+  analyticsMode?: 'legacy' | 'next' | 'duplex';
```

Comment:
Wondering if it would be clearer if the modes we named according to the protocol? e.g.

```suggestion
  analyticsMode?: 'usage analytics' | 'event protocol' | 'duplex';
```